clang-tools-extra/clang-tidy/cppcoreguidelines/MacroUsageCheck.cpp

```diff
@@ -80,6 +80,20 @@ void MacroUsageCheck::warnMacro(const MacroDirective *MD, StringRef MacroName) {
   const MacroInfo *Info = MD->getMacroInfo();
   StringRef Message;
 
+  for (const auto &T : MD->getMacroInfo()->tokens()) {
+    if (T.is(tok::hash)) {
+      return;
+    }
+    if (T.is(tok::identifier)) {
+      StringRef IdentName = T.getIdentifierInfo()->getName();
+      if (IdentName == "__FILE__") {
+        return;
+      } else if (IdentName == "__LINE__") {
+        return;
+      }
+    }
+  }
+
```

Comment:
Won't work for:
```
 #define MyLineAndFile __FILE__, __LINE__
#define logSomething(x) log(MyLineAndFile, x)
```

For me MacroExpands woud need to be used. And even for this scenario, would be good to add some configuration option regex and match to that.

Reply:
If I understand `MacroExpands` macro, it only gets invoked if the macro is used therefore it would need to be called somewhere for the check to trigger. Your code snippet will still error out with `MacroDefined` and `MacroExpands` won't ever be invoked.

Is there a way to use the post-processor inside of `MacroDefined` to expand the definition?

Reply:
Yes, probably easiest option would be simply to add config option like IgnoreMacrosThatUse, or something similar, make it regexp, and then add release note entry that "Macros that use directly an built-in macros are ignored". If you dont want to make that option public, it's fine, it can be some global static vector with strings in a check. Just to avoid bunch of ifs....